hudi-client/hudi-client-common/src/test/java/org/apache/hudi/config/TestHoodieWriteConfig.java

```diff
@@ -140,8 +140,10 @@ public void testAutoConcurrencyConfigAdjustmentWithTableServices(HoodieTableType
             put(ASYNC_CLEAN.key(), "false");
             put(HoodieWriteConfig.AUTO_ADJUST_LOCK_CONFIGS.key(), "true");
           }
-        }), true, true, true, WriteConcurrencyMode.OPTIMISTIC_CONCURRENCY_CONTROL,
-        HoodieFailedWritesCleaningPolicy.LAZY, inProcessLockProviderClassName);
+        }), true, true, true,
```

Comment:
We should revert the changes in this PR to a degree that the auto-adjustment of the lock configs still works for single-writer with async table services.  Right now, auto-adjustment of the lock configs does not work for Deltastreamer with async table services when the metadata table is enabled.

Reply:
If metadata table is enabled, the lock should take effect, because the default lock provider class is `ZookeeperBasedLockProvider`, so at least, the in-progress lock should work.